subsys/net/ip/tcp2.c

```diff
@@ -804,6 +811,11 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		goto next_state;
 	}
 
+	if (!tcp_options_check((th + 1), tcp_options_len)) {
```

Comment:
@rmstoi 

The options check should perhaps be called only if options are present, i.e. tcp_options_len > 0.